### PR TITLE
[resourcedetection] Unify gke and gce detectors and fix gke zone/region detection

### LINKED ISSUE #9682: GKE Resource detection doesn't work well with workload identity
**Describe the bug**

Create a GKE cluster with workload identity enabled.  Try using the GCE resource detector.  It produces error messages because of missing fields.  Try to use the GKE resource detector instead.  Now you are missing important resource information.

The GKE resource detector should set zone/region based on the cluster-location metadata attribute instead.

### PR DESCRIPTION
**Pull request by dashpole**
**Description:**
Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9682

This is modeled after https://github.com/open-telemetry/opentelemetry-go-contrib/pull/2310, which implements resource detection for the go SDK.

It replaces the `gce` and `gke` detectors with a single `gcp` detector.  This eliminates the need to correctly order the two detectors when running on GKE to get correct behavior.  In order to facilitate the migration, i've added warning messages, and ensured that when multiple gcp detectors are used that they are deduplicated.

This adds a new dependency: https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/tree/main/detectors/gcp, which is a library for detecting attributes in GCP environments.  It is [integration tested](https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/e2cce7b04a10906fca4fd38b95ba824a085d61e2/e2e-test-server/endtoendserver/server.go#L153) in real GCP environments.

This adds support for Cloud Run, Cloud Functions (not sure why/how someone would run a collector there, but its possible in theory :)), and App Engine as well.

Removal of 'gke' and 'gce' is tracked in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10348.

**Testing:**

Unit tests added.

**Documentation:**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
 - `jmxreceiver`: Hash the jars provided to JMX Receiver and only allow if they match an approved list (#9687)
 - `jmxreceiver`: Remove properties & groovyscript parameters from JMX Receiver. Add ResourceAttributes & LogLevel parameter to supply some of the removed functionality with reduced attack surface (#9685)
+- `resourcedetectionprocessor`: 'gke' and 'gce' resource detectors are replaced with a single 'gcp' detector (#10347)
```
Comment (pmm-sumo) [inline CHANGELOG.md:27]: This needs to go up since we released `v0.52.0` in the meantime

File: CHANGELOG.md [hunk 2]
```diff
@@ -56,6 +57,7 @@
 - `podmanreceiver`: Container Stats Error structure (#9397)
 - `pkg/stanza`: pipeline.Operators() will return a consistently ordered list of operators whenever possible (#9761)
 - `tanzuobservabilityexporter`: add  error checks for tanzuobservability exporter (#10188)
+- `resourcedetectionprocessor`: GCP resource detector now properly detects zone/region on GKE (#10347)
```
Comment (pmm-sumo) [inline CHANGELOG.md:60]: Needs to go up as well

**Review by pmm-sumo — APPROVED** [review]
There are some trivial Changelog fixes needed but otherwise LGTM

File: CHANGELOG.md (hunk 2 above)
Comment (dashpole) [reply CHANGELOG.md]: done

File: CHANGELOG.md (hunk 1 above)
Comment (dashpole) [reply CHANGELOG.md]: done

**Pull request merged**